src/functions/Coverage.ps1

```diff
@@ -211,54 +211,35 @@ function Resolve-CoverageInfo {
 
     $filePaths = Get-CodeCoverageFilePaths -Paths $resolvedPaths -IncludeTests $includeTests -RecursePaths $recursePaths
 
-    $params = @{
+    $commonParams = @{
         StartLine = $UnresolvedCoverageInfo.StartLine
         EndLine   = $UnresolvedCoverageInfo.EndLine
         Class     = $UnresolvedCoverageInfo.Class
         Function  = $UnresolvedCoverageInfo.Function
     }
 
     foreach ($filePath in $filePaths) {
-        $params['Path'] = $filePath
-        New-CoverageInfo @params
+        New-CoverageInfo @commonParams -Path $filePath
     }
 }
 
 function Get-CodeCoverageFilePaths {
     param (
-        [object]$Paths,
+        [string[]]$Paths,
         [bool]$IncludeTests,
         [bool]$RecursePaths
     )
 
     $testsPattern = "*$($PesterPreference.Run.TestExtension.Value)"
 
-    $filePaths = foreach ($path in $Paths) {
-        $item = & $SafeCommands['Get-Item'] -LiteralPath $path
-        if ($item -is [System.IO.FileInfo] -and ('.ps1', '.psm1') -contains $item.Extension -and ($IncludeTests -or $item.Name -notlike $testsPattern)) {
-            $item.FullName
-        }
-        elseif ($item -is [System.IO.DirectoryInfo]) {
-            $children = foreach ($i in & $SafeCommands['Get-ChildItem'] -LiteralPath $item) {
-                # if we recurse paths return both directories and files so they can be resolved in the
-                # recursive call to Get-CodeCoverageFilePaths, otherwise return just files
-                if ($RecursePaths) {
-                    $i.PSPath
-                }
-                elseif (-not $i.PSIsContainer) {
-                    $i.PSPath
-                }
-            }
-            Get-CodeCoverageFilePaths -Paths $children -IncludeTests $IncludeTests -RecursePaths $RecursePaths
-        }
-        elseif (-not $item.PsIsContainer) {
-            # todo: enable this warning for non wildcarded paths? otherwise it prints a ton of warnings for documentation and so on when using "folder/*" wildcard
-            # & $SafeCommands['Write-Warning'] "CodeCoverage path '$path' resolved to a non-PowerShell file '$($item.FullName)'; this path will not be part of the coverage report."
+    [string[]] $filteredFiles = @(foreach ($file in (& $SafeCommands['Get-ChildItem'] -LiteralPath $Paths -File -Recurse:$RecursePaths)) {
```

Comment:
Use of `-LiteralPath` is key here as `Get-ChildItem -Path c:\test.ps1 -Recurse -File` will recursively look for `test.ps1` while `-LiteralPath` avoids this.

Reply:
I don't think I understand what the difference is. I experimented with it and it returned my child files in both cases.

Reply:
![image](https://github.com/pester/Pester/assets/3436158/c4626297-8c80-4709-97d3-a69cbac6c2f7)

Reply:
I get it now.